sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3937,6 +3937,25 @@ object SQLConf {
     .intConf
     .createWithDefault(20)
 
+  val ARTIFACTS_SESSION_ISOLATION_ENABLED =
+    buildConf("spark.sql.artifact.isolation.enabled")
+      .internal()
+      .doc("When enabled, artifacts (such as JARs, files, archives) added to one SparkSession " +
+        "are isolated from other sessions within the same Spark instance.")
+      .version("4.0.0")
+      .booleanConf
+      .createWithDefault(true)
+
+  val ARTIFACTS_SESSION_ISOLATION_ALWAYS_APPLY_CLASSLOADER =
```

Comment:
Why do we need this flag? It seems that we should be as lazy as possible here.

Reply:
This flag is solely used by Connect, to not break any existing test case and allow it to access artifacts in the "default" workspace before the first `adArtifact` call.

Reply:
Which test is failing? It is a bit weird to me that we need this for connect.

Reply:
This: https://github.com/xupefei/spark/actions/runs/11628160404/job/32382826661
It checks for the error message from the StubClassLoader.